src/pybind/mgr/dashboard/controllers/nvmeof.py

```
@@ -409,24 +429,17 @@ def update(
                         subsystem_nqn=nqn, nsid=int(nsid), new_size=new_size_mib
                     )
                 )
-                if response.status != 0:
-                    return response
```

maybe its because we are doing an early return here?  from the UI, we usually do multiple operations from a single from consuming a single API rather which makes it easier.  Maybe restricting that, I would consider an anti-pattern from what I used to see.

More useful response for the CLI would be an aggregate response of whatever happened right?

---

I thought so  myself but then saw the implementation and got some doubts, so I tried to keep things quite close to how the current implementation works.
I started to think that since this operations are not atomic, we might end up with a partial completion of the update, and from here its not trivial to expect the user to know how to recover from the failure. returning an aggregate response might be the solution but the user's recovery logic in case of an error is complex and error prone itslef, so maybe its not so good as well. 
Also another consideration is that changing the output structure might require changes to the API users code, which is a bit out of my scope (not just because I'm lazy, I am really not familiar with those clients and their code :) )
I think my additions are not making this into a perfect code, but I believe it is at least a step forward in not confusing the user by skipping requested operations or hiding errors, and that perfecting this route will require considerably more effort and decision making.
I hope it makes sense, of course let me know what are your thoughts :)